Sources/PointFree/Account/Cancel.swift

```diff
@@ -37,42 +38,76 @@ let cancelMiddleware =
     <<< requireStripeSubscription
     <<< filter(
       get1 >>> ^\.isRenewing,
-      or: redirect(to: .account(.index), headersMiddleware: flash(.error, "Your subscription is already canceled!"))
+      or: redirect(
+        to: .account(.index),
+        headersMiddleware: flash(.error, "Your subscription is already canceled!")
+      )
     )
     <| cancel
-    >-> redirect(to: .account(.index), headersMiddleware: flash(.notice, "We’ve canceled your subscription."))
 
 let reactivateMiddleware =
   filterMap(require1 >>> pure, or: loginAndRedirect)
     <<< requireStripeSubscription
     <<< filter(
       get1 >>> ^\.cancelAtPeriodEnd,
-      or: redirect(to: .account(.index), headersMiddleware: flash(.error, "Your subscription can’t be reactivated!"))
+      or: redirect(
+        to: .account(.index),
+        headersMiddleware: flash(.error, "Your subscription can’t be reactivated!")
+      )
     )
     <<< requireSubscriptionItem
     <| reactivate
-    >-> redirect(to: .account(.index), headersMiddleware: flash(.notice, "We’ve reactivated your subscription."))
 
 // MARK: -
 
 private func cancel(_ conn: Conn<StatusLineOpen, Tuple2<Stripe.Subscription, Database.User>>)
-  -> IO<Conn<StatusLineOpen, Prelude.Unit>> {
+  -> IO<Conn<ResponseEnded, Data>> {
 
     // TODO: send emails
     return AppEnvironment.current.stripe.cancelSubscription(get1(conn.data).id)
       .run
-      .map(const(conn.map(const(unit))))
```

Comment:
Oops, realized that Stripe failures were getting ignored! Fixes and tests added.